[5.0.0] require() of a non-existing package in a try-catch block is broken

# Bug report

**What is the current behavior?**
require() of a non-existing package in a try-catch block doesn't behave like in Webpack 4.
```
try
{
	require('preact').cloneElement; // when preact not available emits an error
	require('preact'); // when preact not available emits a warning
}
catch(e) {}
```
**What is the expected behavior?**
Webpack should always emit a warning. This worked fine in Webpack 4.

**Other relevant information:**
webpack version: 5.0.0
Node.js version: 12.18.4
Operating System: Linux

https://github.com/siilike/webpack11639
```
$ pnpm i
Packages: +682
Resolving: total 683, reused 683, downloaded 0, done
node_modules/.pnpm/registry.npmjs.org/ejs/2.7.4/node_modules/ejs: Running postinstall script, done in 38ms

devDependencies:
+ webpack 5.0.0
+ webpack-cli 4.0.0

$ node_modules/.bin/webpack-cli ./bug.js
assets by status 176 bytes [cached] 1 asset
./bug.js 93 bytes [built] [code generated]

ERROR in ./bug.js 3:1-31
Module not found: Error: Can't resolve 'preact' in '/data1/tmp/webpacktest'

webpack 5.0.0 compiled with 1 error in 142 ms

$ node_modules/.bin/webpack-cli ./works.js
asset main.js 310 bytes [emitted] [minimized] (name: main)
./works.js 140 bytes [built] [code generated]

WARNING in ./works.js 5:9-26
Module not found: Error: Can't resolve 'preact' in '/data1/tmp/webpacktest'

webpack 5.0.0 compiled with 1 warning in 175 ms
```

--- a/lib/dependencies/CommonJsImportsParserPlugin.js
+++ b/lib/dependencies/CommonJsImportsParserPlugin.js
@@ -238,6 +238,7 @@ class CommonJsImportsParserPlugin {
 					members
 				);
 				dep.asiSafe = !parser.isAsiPosition(expr.range[0]);
+				dep.optional = !!parser.scope.inTry;
 				dep.loc = expr.loc;
 				parser.state.module.addDependency(dep);
 				return true;
@@ -254,6 +255,7 @@ class CommonJsImportsParserPlugin {
 				);
 				dep.call = true;
 				dep.asiSafe = !parser.isAsiPosition(expr.range[0]);
+				dep.optional = !!parser.scope.inTry;
 				dep.loc = expr.callee.loc;
 				parser.state.module.addDependency(dep);
 				parser.walkExpressions(expr.arguments);

--- a/test/cases/runtime/missing-module-exception/index.js
+++ b/test/cases/runtime/missing-module-exception/index.js
@@ -1,7 +1,9 @@
-it("should have correct error code", function() {
+it("should have correct error code", function () {
 	try {
-		require("./fail");
-	} catch(e) {
+		require("./fail-1");
+		require("./fail-2").property;
+		require("./fail-3").property.sub();
+	} catch (e) {
 		expect(e.code).toBe("MODULE_NOT_FOUND");
 	}
 });

--- a/test/cases/runtime/missing-module-exception/warnings.js
+++ b/test/cases/runtime/missing-module-exception/warnings.js
@@ -1,3 +1,5 @@
 module.exports = [
-	[/Module not found/, /Can't resolve '\.\/fail' /]
+	[/Module not found/, /Can't resolve '\.\/fail-1' /],
+	[/Module not found/, /Can't resolve '\.\/fail-2' /],
+	[/Module not found/, /Can't resolve '\.\/fail-3' /]
 ];

